<launch>
<!-- Compare mesh between two meshes -->
    <arg name="visualization" default="true"/>
    <arg name="path" default="/home/jaeyoung/dev/random"/>
    <arg name="groundtruth_mesh" default="/home/jaeyoung/dev/mesh/groundtruth_roi_meshlab.obj"/>
    <arg name="estimated_mesh" default="$(find photogrammetry_evaluations)/dataset/dense/meshed-delaunay.ply"/>

    <node pkg="tf" type="static_transform_publisher" name="world_map" args="0 0 0 0 0 0 world map 10"/>
    <node pkg="photogrammetry_evaluations" type="compare_mesh_node" name="test_mesh" output="screen" required="true">
        <param name="groundtruth_mesh_path" value="$(arg groundtruth_mesh)"/>
        <param name="estimated_mesh_path" value="$(arg estimated_mesh)"/>
        <param name="visualize" value="$(arg visualization)"/>
        <param name="output_path" value="$(arg path)/map_data.csv"/>
    </node>

    <node pkg="grid_map_visualization" type="grid_map_visualization" name="grid_map_visualization" output="screen">
        <rosparam command="load" file="$(find adaptive_viewutility)/launch/visualize.yaml" />
    </node>

    <group if="$(arg visualization)">
        <node type="rviz" name="rviz" pkg="rviz" args="-d $(find photogrammetry_evaluations)/launch/config.rviz" required="true"/>
    </group>
</launch>
